examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,351 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\Examples\AdvancedOperations;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+use GetOpt\GetOpt;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentNames;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentParser;
+use Google\Ads\GoogleAds\Lib\V1\GoogleAdsClient;
+use Google\Ads\GoogleAds\Lib\V1\GoogleAdsClientBuilder;
+use Google\Ads\GoogleAds\Lib\V1\GoogleAdsException;
+use Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder;
+use Google\Ads\GoogleAds\V1\Common\ExpandedDynamicSearchAdInfo;
+use Google\Ads\GoogleAds\V1\Common\ManualCpc;
+use Google\Ads\GoogleAds\V1\Common\WebpageConditionInfo;
+use Google\Ads\GoogleAds\V1\Common\WebpageInfo;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupAdStatusEnum\AdGroupAdStatus;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupCriterionStatusEnum\AdGroupCriterionStatus;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupStatusEnum\AdGroupStatus;
+use Google\Ads\GoogleAds\V1\Enums\AdGroupTypeEnum\AdGroupType;
+use Google\Ads\GoogleAds\V1\Enums\AdvertisingChannelTypeEnum\AdvertisingChannelType;
+use Google\Ads\GoogleAds\V1\Enums\BudgetDeliveryMethodEnum\BudgetDeliveryMethod;
+use Google\Ads\GoogleAds\V1\Enums\CampaignStatusEnum\CampaignStatus;
+use Google\Ads\GoogleAds\V1\Enums\WebpageConditionOperandEnum\WebpageConditionOperand;
+use Google\Ads\GoogleAds\V1\Resources\Ad;
+use Google\Ads\GoogleAds\V1\Resources\AdGroup;
+use Google\Ads\GoogleAds\V1\Resources\AdGroupCriterion;
+use Google\Ads\GoogleAds\V1\Resources\AdGroupAd;
+use Google\Ads\GoogleAds\V1\Resources\Campaign;
+use Google\Ads\GoogleAds\V1\Resources\Campaign\DynamicSearchAdsSetting;
+use Google\Ads\GoogleAds\V1\Resources\CampaignBudget;
+use Google\Ads\GoogleAds\V1\Services\AdGroupCriterionOperation;
+use Google\Ads\GoogleAds\V1\Services\AdGroupOperation;
+use Google\Ads\GoogleAds\V1\Services\AdGroupAdOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignBudgetOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignOperation;
+use Google\Ads\GoogleAds\V1\Services\MutateAdGroupAdsResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateAdGroupsResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateAdGroupCriteriaResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateCampaignBudgetsResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateCampaignsResponse;
+use Google\Protobuf\Int64Value;
+use Google\Protobuf\StringValue;
+
+/**
+ * This example adds a new dynamic search ad (DSA) and webpage targeting criteria for the DSA.
```

Comment:
"criteria" is plural of "criterion". We only create one criterion here so I would change to criterion.

Reply:
This would be because of the golden PR?
Technically, if multiple criteria are allowed, I think it's fine to leave this as is in case we want to add more criteria in the future.
But if not, then definitely this needs a change.

Reply:
I think @PierrickVoulet is right, if we set a single criterion it's correct to use the singular word.

Fixed here and elsewhere.